validation/scenario_generator.py

```diff
@@ -52,7 +52,7 @@ def generate_scenario(start_date_str, end_date_str, raw_df, countries=None, scen
         if start_date < INCEPTION_DATE:
             raise ValueError(f"start_date {start_date} must be on or after inception date {INCEPTION_DATE}")
 
-    ips_df = raw_df[ID_COLS + NPI_COLUMNS]
+    ips_df = raw_df[ID_COLS + NPI_COLUMNS].copy()
```

Comment:
Silencing another pandas warning. It looks like the Dataframes are small here so making a deep copy shouldn't be expensive.

Reply:
Same as above

Reply:
The copy is actually only needed when not filtering by country (the filtering will result in a copy). But I think it's clearer to explicitly make a copy first. 👍